dirsrvtests/tests/suites/schema/schema_csn_replication_test.py

```diff
@@ -0,0 +1,386 @@
+            for user in test_users:
+                user.delete()
+        except Exception as e:
+            log.warning(f"Cleanup warning: {e}")
+
+    request.addfinalizer(cleanup)
+
+    return test_users
+
+
+def wait_for_schema_csn_sync(instance1, instance2, timeout=30):
+    """Wait for schema CSN synchronization"""
+    schema1 = Schema(instance1)
+    schema2 = Schema(instance2)
+
+    for i in range(timeout):
+        csn1 = schema1.get_schema_csn()
+        csn2 = schema2.get_schema_csn()
+
+        if csn1 is not None and csn2 is not None and csn1 == csn2:
+            return True
+
+        if i == 0:
+            log.debug(f"Schema CSN sync in progress: {instance1.serverid} to {instance2.serverid}")
+        time.sleep(1)
+
+    log.error(f"Schema CSN sync failed after {timeout}s: {instance1.serverid} CSN={csn1}, {instance2.serverid} CSN={csn2}")
+    pytest.fail(f"Schema CSN did not sync in {timeout} seconds")
+
+
+def trigger_replication(repl_manager, supplier1, supplier2, test_user):
+    """Trigger replication by modifying a test entry"""
+    test_user.replace('description', f'Replication trigger at {time.time()}')
+    repl_manager.wait_for_replication(supplier1, supplier2)
+
+
+def test_schema_csn_replication_basic_sync(topology_m2, schema_csn_setup):
+    """Test basic schema replication and CSN synchronization
+
+    :id: 8a9b5c42-6d7e-4f3c-a5b2-1e9f8d7c6b5a
+    :setup: Two supplier replication topology with test users
+    :steps:
+        1. Add custom attribute types and object classes to supplier2
+        2. Modify standard schema definitions on supplier2
+        3. Trigger replication from supplier1 to supplier2
+        4. Verify schema CSN synchronization
+    :expectedresults:
+        1. Schema changes are applied successfully
+        2. Replication works correctly
+        3. Schema CSNs are synchronized between suppliers
+    """
+    supplier1 = topology_m2.ms["supplier1"]
+    supplier2 = topology_m2.ms["supplier2"]
+    test_users = schema_csn_setup
+    test_user = test_users[0]
+    repl = ReplicationManager(DEFAULT_SUFFIX)
+
+    log.info('Add custom attribute types and object classes to supplier2')
+    schema2 = Schema(supplier2)
+
+    custom_at = create_custom_attributetype('testAttr', 1)
+    schema2.add_attributetype(custom_at)
+
+    custom_oc = create_custom_objectclass('testClass', 1)
+    schema2.add_objectclass(custom_oc)
+
+    log.info('Trigger replication from supplier1 to supplier2')
+    trigger_replication(repl, supplier2, supplier1, test_user)
+
+    users1 = UserAccounts(supplier1, DEFAULT_SUFFIX)
+    user1 = users1.get(test_user.get_attr_val_utf8('uid'))
+    assert user1.get_attr_val_utf8('description') == test_user.get_attr_val_utf8('description')
+
+    log.info('Verify schema CSN synchronization')
+    wait_for_schema_csn_sync(supplier1, supplier2)
+
+
+def test_schema_csn_after_data_replication(topology_m2, schema_csn_setup):
```

Comment:
IIUC this test does not update the schema. So it relies on the previous test (test_schema_csn_replication_basic_sync) to create the schemaCSN and not sure it can be launched separately